test/TestUtils.py

```diff
@@ -185,3 +187,48 @@ def test_expand_paths_vars(monkeypatch):
     monkeypatch.setenv('TEST_PATH', test_path)
     assert utils.expand_paths_vars(['~']) == [os.path.expanduser('~')]
     assert utils.expand_paths_vars(['$TEST_PATH']) == [test_path]
+
+
+def test_get_yaml_files_no_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.setenv('PATH', '')
+    utils.get_yaml_files(options)
+    msg1, msg2 = capsys.readouterr().out.splitlines()[:2]
+    assert msg1 == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert msg2.startswith('Warning: Failed to locate command:')
+
+
+def test_get_yaml_files_not_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.chdir('/tmp')
+    utils.get_yaml_files(options)
+    msg1, msg2 = capsys.readouterr().out.splitlines()[:2]
+    assert msg1 == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert msg2.startswith('Warning: Failed to discover yaml files to lint using git:')
+
+
+@pytest.mark.parametrize('is_in_git', (True, False))
+def test_get_yaml_files_silent(is_in_git, monkeypatch, capsys, tmp_path):
+    options, _ = cli.get_config([])
+    test_dir = Path(os.path.dirname(os.path.realpath(__file__)))
+    role_sub_dir = Path('roles/test-role')
+    role_dir = test_dir / 'roles/test-role'
+
+    if is_in_git:
+        lint_path = role_dir
+    else:
+        lint_path = tmp_path / role_sub_dir
+        shutil.copytree(role_dir, lint_path)
```

Comment:
What if we tried patching the env var instead?
https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables#_repository_locations
```suggestion
    lint_path = role_dir
    if not is_in_git:
        monkeypatch.setenv('GIT_DIR', '/dev/null')
```

Reply:
Yep, this is much better!